--- ./src/C/mmult.c
+++ ./src/C/mmult.c
@@ -20,6 +20,7 @@
 #include <string.h>
 #include <mpi.h>
 #include <math.h>
+#include <cblas.h>
 
 #define DEFAULT_FN "res_C.mat"
 #define DEFAULT_SIZE 64
@@ -161,7 +161,7 @@
 
   printf("%d: Processing...\n", mr);
   
-  mmult(sz, nproc, mat_a, mat_b, mat_c);
+  cblas_dgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, sz/nproc, sz, sz, 1.0, mat_a, sz, mat_b, sz, 1.0, mat_c, sz);
   
   if(mr == 0)
   {


--- ./src/F90/mmult.F90
+++ ./src/F90/mmult.F90
@@ -107,7 +107,10 @@

   print *,mr,": Processing..."

-  call mmult(sz, nproc, mat_a, mat_b, mat_c)
+  call DGEMM('N','N', sz, sz/nproc, sz, 1.0D0, &
+             mat_b, sz, &
+             mat_a, sz, 1.0D0, &
+             mat_c, sz)

   if(mr==0) then
     print *,mr,": Receiving result matrix..."


--- ./src/make.def
+++ ./src/make.def
@@ -18,6 +18,6 @@
 FC = mpif90

 # Define additional compilation flags
-CFLAGS = -Ofast -g
-LFLAGS =
+CFLAGS = -Ofast -g -I/opt/openblas/0.2.20/include
+LFLAGS = -L/opt/openblas/0.2.20/lib -lopenblas

